src/core/cgroup.c

```diff
@@ -1847,7 +1847,7 @@ CGroupMask unit_get_target_mask(Unit *u) {
 
         own_mask = unit_get_own_mask(u);
 
-        if (own_mask & CGROUP_MASK_BPF_FIREWALL & ~u->manager->cgroup_supported)
+        if ((own_mask & CGROUP_MASK_BPF_FIREWALL & ~u->manager->cgroup_supported) && !FLAGS_SET(u->manager->test_run_flags, MANAGER_TEST_RUN_MINIMAL))
```

Comment:
The change seems not sufficient.
Several code path in `bpf_firewall_supported()` returns `BPF_FIREWALL_UNSUPPORTED` without setting `bpf_firewall_unsupported_reason`. So the assertion can be triggered by PID1, not only in test mode.

Reply:
Maybe, but this at least lets you run analyze with debug logs without crashing, can't the rest be fixed separately?

Reply:
If possible, please.

Reply:
I don't like this approach. The real problem is that `cgroup_supported` is not initialized and/or `bpf_firewall_unsupported_reason` is not initialized. We should figure out why that happens, since those variables are also used in other places. I also don't like the fact that `test_run_flags` is "leaking" into low-level code.

The general idea with `test_run_flags` was that we'd skip parts of the setup. And `manager_setup_cgroup` in fact does conditionalize on `MANAGER_IS_TEST_RUN(m)` in various places, and should set `m->cgroup_supported`. It calls `cg_bpf_mask_supported()`, so `bpf_firewall_supported()` really looks like the culprit here.

Is it enough to to fix `bpf_firewall_supported()` to always set `…_reason` to resolve this?

Reply:
> Is it enough to to fix bpf_firewall_supported() to always set …_reason to resolve this?

No, I'm not even sure it is called in those paths

Reply:
It's not called because manager_setup_cgroup() is not called in test mode - and it shouldn't, as it should not meddle with the system setup

This still seems the right fix to me: it's running in a mode that should not interact with the system

Reply:
I've moved the check inside emit_bpf_firewall_warning